src/NosCore.Controllers/DefaultPacketController.cs

```diff
@@ -292,5 +303,95 @@ public void Pulse(PulsePacket pulsePacket)
                 Session.Disconnect();
             }
         }
-	}
+
+	    /// <summary>
+	    ///     SayPacket
+	    /// </summary>
+	    /// <param name="sayPacket"></param>
+	    public void SayPacket(ClientSayPacket sayPacket)
+	    {
+            //TODO: Add a penalty check when it will be ready
+	        var type = SayColorType.White;
+            Session.Character.MapInstance?.Broadcast(Session, Session.Character.GenerateSay(new SayPacket
+            {
+                Message = sayPacket.Message,
+                Type = type
+            } ), ReceiverType.AllExceptMe);
+	    }
+
+	    /// <summary>
+	    ///     WhisperPacket
+	    /// </summary>
+	    /// <param name="whisperPacket"></param>
+	    public void WhisperPacket(WhisperPacket whisperPacket)
+	    {
+	        try
+	        {
+	            string message = string.Empty;
+	            if (string.IsNullOrEmpty(whisperPacket.Message))
+	            {
+	                return;
+	            }
+
+	            //Todo: review this
+	            string[] messageData = whisperPacket.Message.Split(' ');
+	            string receiverName = messageData[whisperPacket.Message.StartsWith("GM ") ? 1 : 0];
+
+	            for (int i = messageData[0] == "GM" ? 2 : 1; i < messageData.Length; i++)
+	            {
+	                message += $"{messageData[i]} ";
+	            }
+
+	            message = whisperPacket.Message.Length > 60 ? whisperPacket.Message.Substring(0, 60) : message;
+	            message = message.Trim();
+
+                Session.SendPacket(Session.Character.GenerateSpk(new SpeakPacket
+                {
+                    SpeakType = SpeakType.Player,
+                    Message = message
+                }));
+
+	            var speakPacket = Session.Character.GenerateSpk(new SpeakPacket
+	            {
+	                SpeakType = Session.Account.Authority >= AuthorityType.GameMaster ? SpeakType.GameMaster : SpeakType.Player,
+	                Message = message
+	            });
+
+                //Todo: Add a check for blacklisted characters when the CharacterRelation system will be done                
+
+	            int? channel = ServerManager.Instance.GetChannelIdPerAccountName(Session.Account.Name);
```

Comment:
doesnt ask the webapi for this the world is current world so you can just get it on ServerManager just register the world id the servermanager(if not already done somewhere else) should know his id